### PR TITLE
Added unit testing for controllers

### PR DESCRIPTION
**Pull request by OlexG**

### CODE REVIEW

**Comment by OlexG** [comment]
Here is a video of the UI changes. 

https://user-images.githubusercontent.com/36348190/114607413-5d81be00-9c51-11eb-99ff-aa0b5df952e2.mp4

File: client/src/Components/Post/Post.js [hunk 1]
```diff
@@ -1,14 +1,34 @@
 import React from 'react';
 import 'bootstrap/dist/css/bootstrap.css';
+import './styles.css';
```
Comment (irenlian) [inline client/src/Components/Post/Post.js]: Isn't it enough just to import it in App.js? I don't remember well. Could you please double-check?

File: client/src/Components/Post/Post.js [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
 
-const Post = (props) =>
-	(
-		<div className='post card mb-2' style={{ 'width': '70%', 'margin': '1em' }}>
-			<div className='card-body'>
-				<h2>{props.title}</h2>
-				<p>{props.body}</p>
+const Post = (props) => {
+	function swapDropdown () {
+		console.log(document.getElementById(props.id).style.height);
```
Comment (irenlian) [inline client/src/Components/Post/Post.js]:
```suggestion
```

File: client/src/Components/Post/Post.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+		if (document.getElementById(props.id).style.height === '0px' || document.getElementById(props.id).style.height === '') {
```
Comment (irenlian) [inline client/src/Components/Post/Post.js]: I would use `useRef` to get the element.

File: client/src/Components/Post/Post.js [hunk 4]
```diff
@@ -1,14 +1,34 @@
...
+			const element = document.getElementById(props.id);
+			element.classList.add('dropdown-hover');
+			element.style.height = element.scrollHeight + 'px';
```
Comment (irenlian) [inline client/src/Components/Post/Post.js]: I would prefer ES6 Template Strings in such cases.

File: server/errorHandler.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export function errorHandler(err, req, res, next) {
+export function errorHandler (err, req, res, next) {
```
Comment (irenlian) [inline server/errorHandler.ts]:
```suggestion
export function errorHandler(err, req, res, next) {
```

File: server/validation/validateAccessJWT.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import jwt = require('jsonwebtoken');
-export default function validateAccessJWT(req, res, next) {
+export default function validateAccessJWT (req, res, next) {
```
Comment (irenlian) [inline server/validation/validateAccessJWT.ts]:
```suggestion
export default function validateAccessJWT(req, res, next) {
```

File: server/validation/validateRefreshJWT.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import jwt = require('jsonwebtoken');
-export default function validateRefreshJWT(req, res, next) {
+export default function validateRefreshJWT (req, res, next) {
```
Comment (irenlian) [inline server/validation/validateRefreshJWT.ts]:
```suggestion
export default function validateRefreshJWT(req, res, next) {
```

File: tests/mocks/mockInitDB.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+export default {
+	initManagers: function () {
+		class PostManager {
+			async getPost (postId: number): Promise<string> {
```
Comment (irenlian) [inline tests/mocks/mockInitDB.ts]: I would definitely go with `jest.fn().mockResolvedValue()` or `jest.fn().mockImplementation()` to have better control in tests for all of the methods.

File: tests/postController.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+		const req = {
+			query: {
+				number: 0,
+				page: 0
+			}
+		};
+		const resSpy = sinon.spy();
+		const res = {
+			send: resSpy
+		};
```
Comment (irenlian) [inline tests/postController.ts]: Just assumption that you could move this to `beforeEach` section. What do you think?

**Review by irenlian — COMMENTED** [review]
Wow, awesome UI changes! And a lot of tests as well.😍👏

My recommendation is just to separate such changes in different PRs to have better tracking of updates.
And you might want to try to set up the coverage analytics for tests to get the visualization of your progress.😏

File: client/src/Components/Post/Post.js (hunk 1 above)
Comment (OlexG) [reply client/src/Components/Post/Post.js]: I thought that css imported here was local. I guess it need to make it into a module

File: tests/mocks/mockInitDB.ts (hunk 8 above)
Comment (OlexG) [reply tests/mocks/mockInitDB.ts]: Sure, I will replace them with sinon.stubs which are similar

File: tests/postController.ts (hunk 9 above)
Comment (OlexG) [reply tests/postController.ts]: Yes it looks I can for postController

**Review by irenlian — APPROVED** [review]

**Pull request merged**